### PR TITLE
Summary route for Thiago to consume

### PR DESCRIPTION
**Pull request by mathcolo**
@austinjpaul You may be at least moderately happier with this :p

This makes a route `/statistics` that returns a JSON object for @thiteixeira to consume. Supports only max for now. Looks like this:
```
{
  "Red": {
    "max": {
      "value": 0,
      "on_date": "2022-05-20"
    }
  },
  "Orange": {
    "max": {
      "value": 0,
      "on_date": "2022-05-20"
    }
  }
}
```

### CODE REVIEW

File: server/history/statistics.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        for route in ROUTES:
+            CACHE[route] = summarize(route)
+
+    return CACHE
+
+
+def summarize(route):
+    csv_buffer = download_history(route).strip().splitlines(True)
+    reader = csv.reader(csv_buffer)
+    next(reader, None)  # Skip the csv header
```
Comment (austinjpaul) [inline server/history/statistics.py:33]: is this really the preferred way of doing this?

File: server/history/statistics.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    max = 0
+    max_date = None
+
+    for date, count in reader:
+        if int(count) >= max:
+            max = int(count)
+            max_date = date
```
Comment (austinjpaul) [inline server/history/statistics.py:41]: This is okay for now. If we were to do more number-crunching here, i'd highly recommend pandas over csv reader.

File: server/history/statistics.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    key = S3_KEY_NEW_TRAINS.format(route)
+    return download(BUCKET, key, compressed=False)
+
+
+def get_cached_summaries():
+    global CACHE
+    global CACHE_LAST_REFRESH_DAY
+    now = datetime.datetime.utcnow()
+    todays_data_upload = now.replace(hour=10, minute=6, second=0, microsecond=0)
+    if CACHE_LAST_REFRESH_DAY is None or (CACHE_LAST_REFRESH_DAY != now.date() and now > todays_data_upload):
```
Comment (austinjpaul) [inline server/history/statistics.py:22]: first condition should maybe check the CACHE itself and not just the day, just in case? idk

**Review by austinjpaul — APPROVED** [review]
Looks alright to me! 
I assume there are other metrics we'd want to count? 

I think I originally assumed nginx would serve the file more directly, but since we need to do some processing, i like this approach. Wonder what @idreyn thinks...

File: server/history/statistics.py (hunk 1 above)
Comment (mathcolo) [reply server/history/statistics.py:33]: Well, pandas is probably the preferred way :p But besides that, it's either this or holding a counter for `reader` and testing for count == 0 :/

Comment (idreyn) [reply server/history/statistics.py:33]: passes the smell test for me!

File: server/history/statistics.py (hunk 3 above)
Comment (mathcolo) [reply server/history/statistics.py:22]: oh true, cannot hurt to be explicit about that.

**Pull request merged**